liblouis/compileTranslationTable.c

```diff
@@ -1612,11 +1611,10 @@ compileSwap(FileInfo *nested, TranslationTableOpcode opcode,
 	} else {
 		if (!compileSwapDots(nested, &replacements, &ruleDots)) return 0;
 	}
-	if (!addRule(nested, opcode, &ruleChars, &ruleDots, 0, 0, &ruleOffset, newRule,
```

Comment:
Should the API of `addRule` be changed? Is there ever a case where you pass a non-null value of `newRule`?

Reply:
There are some cases where a non-null `newRule` is passed, see for example in the code that handles `CTO_RepWord`. The returned pointer is used to set the `nocross` field of the rule.

That said, there is also the `newRuleOffset` variable which carries the same information as `newRule`, because a rule can always be computed from an offset as `table->ruleArea[ruleOffset]`. In fact the offset is more reliable because pointers break when the table is reallocated. So maybe it is better to pass around offsets as much as possible and avoid passing around pointers (everywhere, not only in this function).